packages/react-components/src/components/UserGuide/types.ts

```diff
@@ -0,0 +1,36 @@
+import { CSSProperties } from 'react';
+
+import { Placement } from '@floating-ui/react';
+
+export type CursorTiming = 'fast1' | 'fast2' | 'moderate1' | 'moderate2';
+
+export interface IUserGuide {
+  /**
+   * The class name for the floating container
+   */
+  className?: string;
+  /**
+   * The CSS properties for the highlighted element
+   */
+  elementStyles?: CSSProperties;
+  /**
+   * The position for the floating element which sets the cursor position
+   */
+  cursorPosition?: Placement;
+  /**
+   * The timing for the floating element transition
+   */
+  cursorTiming?: CursorTiming;
+  /**
+   * The id of the element to highlight
+   */
+  parentElementName?: string;
```

Comment:
Shouldn't `parentElementName`, `cursorPosition` and `isVisible` be required props? 🤔

Reply:
Good point, but we currently working on the "first step" and "last step" features, which will not require the element to highlight, as well as other positioning props.